### PR TITLE
Admin: Prevent user from deleting user's current/active organization

### LINKED ISSUE #17464: Grafana and org admin deleting its current org leaves admin with non-existent current org and unable to set org roles
**What happened**:
Grafana Admin `admin`creates a new organization `neworg1`. That org is now its "Current" org. This user now deletes `neworg1`, then logs out of grafana and logs in again.
`admin` now cannot change org roles despite being Grafana Admin and seeing the users in the org. The column for selecting the user roles for the organization is simply missing.

Looking at `grafana.db` in table `user`, `admin` still has the `org_id` of the now-deleted organization.

**What you expected to happen**:
Either deleting the organization should be prohibited (not preferred) or the admin should be assigned to another (the default?) organization.

**How to reproduce it (as minimally and precisely as possible)**:
See above.

**Environment**:
- Grafana version: 6.1.4
- Data source type & version: Elasticsearch 6
- OS Grafana is installed on: Ubuntu in Docker
- User OS & Browser: macos, Firefox
- Grafana plugins: -
- Others: AuthProxy with auto signup enabled

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:
This PR would add a check before deleting the org. That a user wouldn't be able to delete their own org.

**Which issue(s) this PR fixes**:

Fixes #17464

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/api/org.go [hunk 1]
```diff
@@ -151,7 +151,23 @@ func updateOrgAddressHelper(form dtos.UpdateOrgAddressForm, orgID int64) respons
 }
 
 // GET /api/orgs/:orgId
-func DeleteOrgByID(c *models.ReqContext) response.Response {
+func (hs *HTTPServer) DeleteOrgByID(c *models.ReqContext) response.Response {
```
Comment (marefr) [inline pkg/api/org.go:154]: Checking if `c.OrgId` equals `c.ParamsInt64(":orgId")` should be enough, right?

**Review by marefr — CHANGES_REQUESTED** [review]
See comment

File: pkg/api/org.go (hunk 1 above)
Comment (idafurjes) [reply pkg/api/org.go:154]: yes, this is much more simpler

File: pkg/api/org.go [hunk 2]
```diff
@@ -152,6 +152,11 @@ func updateOrgAddressHelper(form dtos.UpdateOrgAddressForm, orgID int64) respons
 
 // GET /api/orgs/:orgId
 func DeleteOrgByID(c *models.ReqContext) response.Response {
+	// before deleting an org, check if user does not belong to the current org
+	if c.OrgId == c.ParamsInt64(":orgId") {
+		return response.Error(403, "Can not delete org for current user", nil)
```
Comment (marefr) [inline pkg/api/org.go:157]: I think 400 makes more sense. 403 implies that the user don't have permission to access this resource. WDYT?

```suggestion
		return response.Error(400, "Can not delete org for current user", nil)
```

File: pkg/api/org.go [hunk 3]
```diff
@@ -152,6 +152,11 @@ func updateOrgAddressHelper(form dtos.UpdateOrgAddressForm, orgID int64) respons
 
 // GET /api/orgs/:orgId
 func DeleteOrgByID(c *models.ReqContext) response.Response {
+	// before deleting an org, check if user does not belong to the current org
+	if c.OrgId == c.ParamsInt64(":orgId") {
```
Comment (marefr) [inline pkg/api/org.go:156]:
```suggestion
	orgID := c.ParamsInt64(":orgId")
	
	// before deleting an org, check if user does not belong to the current org
	if c.OrgId == orgID {
```

File: pkg/api/org.go [hunk 4]
```diff
@@ -152,6 +152,11 @@ func updateOrgAddressHelper(form dtos.UpdateOrgAddressForm, orgID int64) respons
...
+		return response.Error(403, "Can not delete org for current user", nil)
+	}
+
 	if err := bus.Dispatch(&models.DeleteOrgCommand{Id: c.ParamsInt64(":orgId")}); err != nil {
```
Comment (marefr) [inline pkg/api/org.go:160]:
```suggestion
	if err := bus.Dispatch(&models.DeleteOrgCommand{Id: orgId}); err != nil {
```

**Review by marefr — APPROVED** [review]
LGTM but added some suggestions

**Comment by marefr** [comment]
Your latest commit seems to be related to #16728? Would suggest handling that in a separate PR. 

Would be nice with a reference in PR description of `Fixing #17464` so that when this PR is merged the issue is automatically closed

**Comment by idafurjes** [comment]
no problem, I can roll back

**Pull request merged**